### PR TITLE
Add Arcade.Sdk packages

### PR DESCRIPTION
**Pull request by alexperovich**
This change adds project and traversal Sdks as part of the Arcade Sdk
I also updated the repository build files to use the new Sdks

Right now I have referenced the Arcade Sdk targets files directly, once
the shape of the props/targets files has been approved I will publish
bootstrapping versions of the files into the arcade feed and reference
them as packages.

### CODE REVIEW

**Comment by rainersigwald** [comment]
What does the Arcade traversal SDK do that [`Microsoft.Build.Traversal`](https://github.com/Microsoft/MSBuildSdks/tree/master/src/Traversal) doesn't? @jeffkl might be willing to add options if we need to.

**Comment by alexperovich** [comment]
The Arcade Traversal sdk contains the publish and eventually sign phases of the repository build. The part I had to add that should probably be in Microsoft.Build.Traversal is the Pack target hookup.

File: src/Microsoft.DotNet.Arcade.Sdk/RepoLayout.props [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  <PropertyGroup>
+    <BuildArtifactDirectory Condition="'$(BuildArtifactDirectory)' == ''">$(RepositoryRoot)artifacts\</BuildArtifactDirectory>
+    <ProjectRelativePath>$([MSBuild]::MakeRelative('$(RepositoryRoot)', '$(MSBuildProjectDirectory)\$(MSBuildProjectName)\'))</ProjectRelativePath>
+
+    <BaseIntermediateOutputPath>$(BuildArtifactDirectory)obj\$(ProjectRelativePath)</BaseIntermediateOutputPath>
+    <BaseOutputPath>$(BuildArtifactDirectory)bin\$(ProjectRelativePath)</BaseOutputPath>
+
+    <PackageOutputPath>$(BuildArtifactDirectory)packages\</PackageOutputPath>
+  </PropertyGroup>
+</Project>
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/RepoLayout.props]: Why is this not an exact copy of RepoLayout from RepoToolset?

File: src/Microsoft.DotNet.Arcade.Sdk/ProjectDefaults.props [hunk 2]
```diff
@@ -0,0 +1,42 @@
+<Project>
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/ProjectDefaults.props]: The same question here: why not take the file as is from RepoToolset?

File: src/Microsoft.DotNet.Arcade.Sdk/Package.props [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <!-- Temporary Versioning until @JohnTortugo finishes the versioning package -->
+    <VersionSuffix>$([System.DateTime]::Now.Ticks)</VersionSuffix>
+    <VersionSuffix>$(VersionSuffix.Substring(0, 10))</VersionSuffix>
+    <NoWarn>NU5105</NoWarn>
+  </PropertyGroup>
+
+  <Import Project="$(MSBuildThisFileDirectory)RepoLayout.props" />
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Sdk/Package.props]: >$(MSBuildThisFileDirectory) [](start = 19, length = 27)

This is not necessary. Import paths are relative to the containing file.

File: src/Microsoft.DotNet.Arcade.Traversal/Publish.targets [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    <ItemGroup>
+      <PackagesToPublish Include="$(PackageOutputPath)*.nupkg"/>
+    </ItemGroup>
+
+    <PushToBlobFeed
+      ExpectedFeedUrl="$(BlobFeedUrl)"
+      AccountKey="$(BlobFeedKey)"
+      ItemsToPush="@(PackagesToPublish)"/>
+  </Target>
+</Project>
```
Comment (tmat) [inline src/Microsoft.DotNet.Arcade.Traversal/Publish.targets]: Why is this here? 

See [https://github.com/dotnet/roslyn-tools/blob/master/src/RepoToolset/tools/Publish.proj](https://github.com/dotnet/roslyn-tools/blob/master/src/RepoToolset/tools/Publish.proj)

**Comment by tmat** [comment]
>   </ItemGroup>

Why have these references conditional on TargetFramework?

---
Refers to: src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj:65 in 45d8b74. [](commit_id = 45d8b74d8672e72cc7e4bb06f31bb053fe3e07f7, deletion_comment = False)

File: src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj [hunk 5]
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project Sdk="RoslynTools.RepoToolset">
+<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <TargetFrameworks>netstandard1.5;$(NetFxTfm)</TargetFrameworks>
```
Comment (tmat) [inline src/Microsoft.DotNet.Build.Tasks.Feed/Microsoft.DotNet.Build.Tasks.Feed.csproj]: >netstandard1.5 [](start = 22, length = 14)

let's just build for `netstandard2.0` and `net461`

File: eng/Versions.props [hunk 6]
```diff
@@ -21,11 +21,4 @@
     <MicrosoftBuildFrameworkVersion>15.1.548</MicrosoftBuildFrameworkVersion>
     <MicrosoftBuildUtilitiesCoreVersion>15.1.548</MicrosoftBuildUtilitiesCoreVersion>
   </PropertyGroup>
-
-  <PropertyGroup>
-    <RestoreSources>
-      $(RestoreSources);
-      https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json;
-    </RestoreSources>
-  </PropertyGroup>
```
Comment (tmat) [inline eng/Versions.props]: Why are you replacing this with nuget.config. 
There is a reason why we use RestoreSources in RepoToolset and not nuget.config -- the orchestrated build and source build need to be able to override the feeds.

File: eng/common/build.ps1 [hunk 7]
```diff
@@ -1,17 +1,14 @@
...
   [string] $verbosity = "minimal",
   [switch] $restore,
-  [switch] $deployDeps,
   [switch] $build,
   [switch] $rebuild,
-  [switch] $deploy,
   [switch] $test,
-  [switch] $integrationTest,
   [switch] $sign,
   [switch] $pack,
```
Comment (tmat) [inline eng/common/build.ps1]: Please update this to the latest version from RepoToolset-based repository, such as SymReader. 

You are removing features that other repos need.

File: eng/common/build.ps1 [hunk 8]
```diff
@@ -99,29 +76,60 @@ function InstallNativeTools {
...
+    $targets += "Test"
+  }
+
+  if ($pack) {
+    $targets += "Pack"
   }
 
-  & $DotNetExe msbuild $ToolsetBuildProj /m /nologo /clp:Summary /warnaserror /v:$verbosity $logCmd /p:Configuration=$configuration /p:RepoRoot=$RepoRoot /p:Projects=$projects /p:Restore=$restore /p:DeployDeps=$deployDeps /p:Build=$build /p:Rebuild=$rebuild /p:Deploy=$deploy /p:Test=$test /p:IntegrationTest=$integrationTest /p:Sign=$sign /p:Pack=$pack /p:CIBuild=$ci /p:RestorePackagesPath=$NuGetPackageRoot /p:NuGetPackageRoot=$NuGetPackageRoot $properties
+  if ($sign) {
+    $targets += "Sign"
```
Comment (tmat) [inline eng/common/build.ps1]: You are reimplementing features that already exist in RepoToolset, just slightly differently. This is a waste of effort.

**Comment by tmat** [comment]
Instead of rolling out a completely new SDK, I'd suggest to keep the Aracade repo on RepoToolset and upgrade it to the latest version. If there are any issues in the latest RepoToolset version that are blocking some work in Arcade let me know (file an issue in Arcade or RoslynTools repo) and let's address them in the toolset.
